### PR TITLE
Replace opscode from folder names, hardcodes of chef-server.rb, chef-server-running.json, chef-secrets.json

### PR DESCRIPTION
**Pull request by aleksey-hariton**
Replacing Chef trademarks where possible.
Related PR`s:
* https://github.com/chef/chef/pull/12513 # Chef 18
* https://github.com/chef/chef/pull/12519 # Chef 17
* https://github.com/chef/chef/pull/12520 # Chef 16
* https://github.com/chef/chef/pull/12539 
* https://github.com/chef/chef/pull/12540 
* https://github.com/chef/chef/pull/12541 
* https://github.com/chef/chef_backup/pull/42
* https://github.com/chef/omnibus-ctl/pull/83
* https://github.com/chef/omnibus-ctl/pull/84

### Description

This changes required to make open-sourcing of ChefServer possible (Cinc project).

### CODE REVIEW

File: omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb [hunk 1]
```diff
@@ -78,11 +81,11 @@
 # The Chef User that services run as
 ####
 # The username for the chef services user
-default['private_chef']['user']['username'] = 'opscode'
+default['private_chef']['user']['username'] = ChefUtils::Dist::Org::LEGACY_CONF_DIR
```
Comment (marcparadise) [inline omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb:84]: this is a username, but we're using a dirname const. If we want to change the default username, this should have its own const.

Comment (ramereth) [reply omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb:84]: @marcparadise Would you like us to add something like `ChefUtils::Dist::Server::USERNAME` to chef-utils?

**Comment by lbakerchef** [comment]
At a minimum I think we need to see that the output (it can be snipped) of `chef-server-ctl test` or even `chef-server-ctl test --all` is successful.  [NOTE: I tried to run an adhoc and planned to run umbrella after, but these apparently don't play nice with forks]

File: omnibus/files/server-ctl-cookbooks/infra-server/templates/default/pivotal.rb.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "tempfile"
 require "veil"
 
-secrets_file = ENV['SECRETS_FILE'] || "/etc/opscode/private-chef-secrets.json"
+secrets_file = ENV['SECRETS_FILE'] || "/etc/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/private-chef-secrets.json"
```
Comment (ramereth) [inline omnibus/files/server-ctl-cookbooks/infra-server/templates/default/pivotal.rb.erb:4]:
```suggestion
secrets_file = ENV['SECRETS_FILE'] || "/etc/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/private-#{ChefUtils::Dist::Infra::SHORT}-secrets.json"
```

File: omnibus/files/server-ctl-cookbooks/infra-server/libraries/private_chef.rb [hunk 3]
```diff
@@ -334,7 +334,7 @@ def gen_api_fqdn
 
     # TODO: 2017-02-28 mp:  configurable location:
     def secrets_json
-      '/etc/opscode/private-chef-secrets.json'
+      "/etc/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/private-chef-secrets.json"
```
Comment (ramereth) [inline omnibus/files/server-ctl-cookbooks/infra-server/libraries/private_chef.rb:337]:
```suggestion
      "/etc/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/private-#{ChefUtils::Dist::Infra::SHORT}-secrets.json"
```

**Review by ramereth — COMMENTED** [review]
This matches what we have on the Cinc side FWIW ...

File: omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb (hunk 1 above)
Comment (aleksey-hariton) [reply omnibus/files/server-ctl-cookbooks/infra-server/attributes/default.rb:84]: @marcparadise if this works for you - https://github.com/chef/chef/pull/12513/files

File: src/chef-server-ctl/lib/chef_server_ctl/config.rb [hunk 4]
```diff
@@ -26,7 +26,7 @@ module Config
     DOC_PATENT_MSG = <<-DOC.freeze
 
 Documentation: #{ChefUtils::Dist::Server::SERVER_DOCS}
-Patents:       #{ChefUtils::Dist::Org::PATENTS}
+Patents:       #{ChefUtils::Dist::Org::PATENT/etc/opscode/S}
```
Comment (ramereth) [inline src/chef-server-ctl/lib/chef_server_ctl/config.rb]: This looks wrong.

Comment (aleksey-hariton) [reply src/chef-server-ctl/lib/chef_server_ctl/config.rb]: Fixed.

**Comment by jashaik** [comment]
Pipeline Information - 
Adhoc - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3947
Umbrella -

**Comment by marcparadise** [comment]
Maybe that failure will be fixed if we update the Chef gem? The const is present: 

https://github.com/chef/chef/blob/main/chef-utils/lib/chef-utils/dist.rb#L115

**Comment by ramereth** [comment]
> Maybe that failure will be fixed if we update the Chef gem? The const is present:
> 
> https://github.com/chef/chef/blob/main/chef-utils/lib/chef-utils/dist.rb#L115

Yes, should we go ahead and bump it in this PR or do it in a different PR? The latest 16 release should include the constant (`v16.17.51`)

**Comment by ramereth** [comment]
@aleksey-hariton @lbakerchef @marcparadise @jashaik I have created #3205 to replace this PR which should have the final fixes (hopefully). This is created from our shared repo which will make it easier for our group to push any additional fixes to.

**Comment by aleksey-hariton** [comment]
Closing in favor to https://github.com/chef/chef-server/pull/3205

**Pull request closed without merging**